dali/imgcodec/decoders/tiff_libtiff.cc

```diff
@@ -82,11 +75,7 @@ class DecoderHelper {
 
  private:
   std::shared_ptr<InputStream> stream_;
-  InputKind kind_;
-
-  // Used only if kind_ == InputKind::HostMemory
-  const void *buffer_;
-  size_t buffer_size_;
+  ImageSource *in_;
```

Comment:
Hmm, now that I realized - we need to guarantee that the ImageSource is kept alive when the Helper is alive. Is this true? Maybe it was better to copy those properties?

Sorry for the confusion.

Reply:
`ImageSource` pointer is passed to `Decode` as an argument. The Helper is constructed in `Decode` and its lifetime is limited to `Decode`'s scope. Isn't that enough?

Reply:
Yeah, it is safe. This helper class is just meant to be used within the `Decode` scope